packages/rum-core/src/performance-monitoring/transaction.js

```diff
@@ -132,8 +134,18 @@ class Transaction extends SpanBase {
     const metadata = getPageMetadata()
     this.addContext(metadata)
 
-    this._adjustStartToEarliestSpan()
-    this._adjustEndToLatestSpan()
+    /**
+     * For page load transactions, we have to adjust the transaction start/end
+     * depending on the navigation & resource timing spans.
+     *
+     * Doing it here would exclude the navigation and resource timing spans since
+     * transaction.end might be lesser than those spans end and
+     * it would be considered as invalid span
+     */
+    if (this.type !== PAGE_LOAD) {
+      this._adjustStartToEarliestSpan()
+      this._adjustEndToLatestSpan()
```

Comment:
We still call these two in the `capture-navigation-timing.js`, not sure why this would solve the issue.
In the capture-navigation-timing we just add the spans and then adjust the start/end of the transaction to be compatible with the spans that were added for resource and navigation.

Reply:
The main point is to do it only in https://github.com/elastic/apm-agent-rum-js/blob/c162a1252f034c2ec59d5c5ce349412857d817aa/packages/rum-core/src/performance-monitoring/capture-hard-navigation.js#L217-L218 for the page-load transaction after we have calculated the nav timing spans. 

Doing it on the `transacton.js` prevents the nav/res timing spans from getting added which is the issue that we are fixing.